src/main/scala/tech/cryptonomic/conseil/tezos/repositories/SlickRepositories.scala

```diff
@@ -0,0 +1,372 @@
+import tech.cryptonomic.conseil.tezos.Tables.{
+  Accounts,
+  AccountsCheckpoint,
+  AccountsCheckpointRow,
+  AccountsRow,
+  BalanceUpdates,
+  BalanceUpdatesRow,
+  Ballots,
+  BallotsRow,
+  Blocks,
+  BlocksRow,
+  DelegatedContracts,
+  DelegatedContractsRow,
+  Delegates,
+  DelegatesCheckpoint,
+  DelegatesCheckpointRow,
+  DelegatesRow,
+  Fees,
+  FeesRow,
+  OperationGroups,
+  OperationGroupsRow,
+  Operations,
+  OperationsRow,
+  Proposals,
+  ProposalsRow,
+  Rolls,
+  RollsRow
+}
+import tech.cryptonomic.conseil.generic.chain.DataTypes.{Aggregation, OutputType, Predicate, QueryOrdering}
+import tech.cryptonomic.conseil.generic.chain.DataTypes.OutputType.OutputType
+import tech.cryptonomic.conseil.metadata.repositories.MetadataRepository
+
+import cats.Applicative
+import slick.jdbc.PostgresProfile.api._
+import slickeffect.implicits._
+import scala.concurrent.ExecutionContext
+
+object SlickRepositories {
+
+  /** Sanitizes predicate values so query is safe from SQL injection */
+  def sanitizePredicates(predicates: List[Predicate]): List[Predicate] =
+    predicates.map { predicate =>
+      predicate.copy(set = predicate.set.map(field => sanitizeForSql(field.toString)))
+    }
+
+  /** Sanitizes string to be viable to paste into plain SQL */
+  def sanitizeForSql(str: String): String = {
+    val supportedCharacters = Set('_', '.', '+', ':', '-', ' ')
+    str.filter(c => c.isLetterOrDigit || supportedCharacters.contains(c))
+  }
+}
+
+class SlickRepositories(implicit ec: ExecutionContext) {
+  import SlickRepositories._
+  import DatabaseConversions._
+
+  /**
+    * Removes  data from a generic checkpoint table
+    * @param selection limits the removed rows to those
+    *                  concerning the selected elements, by default no selection is made.
+    *                  We strictly assume those keys were previously loaded from the checkpoint table itself
+    * @param tableQuery the slick table query to identify which is the table to clean up
+    * @param tableTotal an action needed to compute the number of max keys in the checkpoint
+    * @param applySelection used to filter the results to clean-up, using the available `selection`
+    * @return the database action to run
+    */
+  private[this] def cleanCheckpoint[PK, Row, T <: Table[Row], CheckpointTable <: TableQuery[T]](
+      selection: Option[Set[PK]],
+      tableQuery: CheckpointTable,
+      tableTotal: DBIO[Int],
+      applySelection: (CheckpointTable, Set[PK]) => Query[T, Row, Seq]
+  )(implicit ec: ExecutionContext): DBIO[Int] =
+    selection match {
+      case Some(pks) =>
+        for {
+          total <- tableTotal
+          marked = if (total > pks.size) applySelection(tableQuery, pks)
+          else tableQuery
+          deleted <- marked.delete
+        } yield deleted
+      case None =>
+        tableQuery.delete
+    }
+
+  implicit val metadataRepository = new MetadataRepository[DBIO, String, String, String] {
```

Comment:
you can copy javadocs for this implementation

Reply:
Sorry, I don't get what you mean

Reply:
there are no javadocs in your implementation of `MetadataRepository`, but I think they could be useful

Reply:
I put docs in the trait: https://github.com/Cryptonomic/Conseil/pull/437/files#diff-ac72c69b299af611e9860531b0d02d9bR8

I didn't add any code comments here for any of the actual implementations... you're suggesting something like

```scala
/** slick implementation of MetadataRepository[DBIO] */
```

and so on?

Reply:
yes, basically what I had in mind was using intellij option to copy comments for implementation - but I forgot that you don't use intellij 😄

Reply:
I can fake it

Reply:
did it